scripts: west_commands: fix ZEPHYR_BASE env check

paulober opened the pull request:
`detect_version()` in the sdk west command used `os.environ["ZEPHYR_BASE"]`, which raises a `KeyError` when the variable is absent, preventing the fallback path from running. Switch to `os.environ.get("ZEPHYR_BASE")` and use `Path(__file__).resolve()` for the fallback so the path is absolute/canonical.

(@tdewey-rpi @will-v-pi)

pdgendt commented:
Please take a look at the compliance issue, you can't use a hidden e-mail address when contributing to Zephyr.

paulober committed:
scripts: west_commands: fix ZEPHYR_BASE env check

detect_version() used os.environ["ZEPHYR_BASE"],
which raises a KeyError when the variable is
absent, preventing the fallback path from running.
Switch to os.environ.get("ZEPHYR_BASE") and use
Path(__file__).resolve() for the fallback so the
path is absolute/canonical.

Signed-off-by: Paul Oberosler <paul.oberosler@raspberrypi.com>
